var/spack/repos/builtin/packages/vtk/package.py

```diff
@@ -30,22 +31,38 @@ class Vtk(CMakePackage):
     variant('osmesa', default=False, description='Enable OSMesa support')
     variant('python', default=False, description='Enable Python support')
     variant('qt', default=False, description='Build with support for Qt')
+    variant('tk', default=False, description='Build with support for Tk')
+    variant('tkinter', default=False, description='Support tkinter')
     variant('xdmf', default=False, description='Build XDMF file support')
     variant('ffmpeg', default=False, description='Build with FFMPEG support')
     variant('mpi', default=True, description='Enable MPI support')
 
     # Haru causes trouble on Fedora and Ubuntu in v8.1.1
     # See https://bugzilla.redhat.com/show_bug.cgi?id=1460059#c13
+    # probably related: https://github.com/libharu/libharu/pull/157
     variant('haru', default=True, description='Enable libharu')
 
     patch('gcc.patch', when='@6.1.0')
 
-    # At the moment, we cannot build with both osmesa and qt, but as of
-    # VTK 8.1, that should change
-    conflicts('+osmesa', when='+qt')
+    # Apparently there are problems
+    # with mesa/offscreen-rendering in older versions of VTK
+    conflicts('+osmesa', when='@:8.1 +qt')
```

Comment:
Given the nature of vtk and rendering back ends, I think we should have osmesa, (also egl soon), qt, and tk as mutually exclusive.